librz/arch/isa/msp430/msp430_il.c

```diff
@@ -0,0 +1,199 @@
+        old_carry
+    );
+
+    return OR(
+        AND(old_sr_value, UN16(1 << 8)), // zero out position 8, leave all else the same
+        SHIFTL(IL_FALSE, UNSIGNED(16, v_flag_value), UN8(8)) // OR with the new value for the bit 
+    ); 
+}
+
+RzILOpBitVector *update_sr_c_flag(const RzILOpBool *new_carry, const RzILOpBitVector *old_sr_value) {
+    return MSP430_SETR(MSP430_SR, OR(
+        AND(old_sr_value, UN16(1)), // zero out the 0th position (i.e. the carry bit)
+        UNSIGNED(16, new_carry)    // OR with the new carry 
+    ));
+}
+
+RzILOpEffect *rz_msp430_lift_single_operand_instr(RZ_NONNULL RzAnalysis *analysis, const Msp430Instruction *op) {
+	switch (op->iopcode) {
+    // rotation is just a shift with filling
+    case MSP430_RRC: {
+        // 1- get the carry (to use later as the filling for the MSB of the operand)
+        RzILOpBitVector *sr = MSP430_GETR(MSP430_SR);
+        RzILOpBool *carry = LSB(sr);
+        // 2- get LSB of the operand (register, memory location, ...)
+        RzILOpBitVector *operand = get_destination(op);
+        RzILOpBool *lsb = LSB(operand);
+        // 3- Perform the Rotate Right through Carry operation: 
+        //      a- Shift the operand by 1 to the right and fill with carry
+        RzILOpBitVector *result = SHIFTR(carry, operand, U8(1));
+        
+        // ...  b- and set the operand to the value of the previous computation 
+        RzILOpEffect *set_operand = set_destination(op, operand, result);
+        // ...  c- and finally set the flags: NZ as usual, v espeically for RCC, and the carry flag from the discarded LSB
+        RzILOpBitVector *sr_1 = update_sr_nz_flags(result, sr);
+        RzILOpBitVector *sr_2 = update_sr_v_flag_rcc(operand, carry, sr_1);
+        RzILOpBitVector *sr_new = update_sr_c_flag(lsb, sr_2);
+
+        return SEQ2(set_operand, MSP430_SETR(MSP430_SR,sr_new));
+    }
+
+    case MSP430_SWPB: {
+        // 1- get lower byte and upper byte of the operand
+        BVPair low_high = get_destination_destructured(op);
+        RzILOpBitVector *low_byte = low_high.first;
+        RzILOpBitVector *high_byte = low_high.second;
+
+        // 2- append them in reverse order
+        RzILOpBitVector *result = APPEND(low_byte, high_byte);
+
+        // 3- set them (flags aren't affected)
+        return set_destination(op, NULL, result);
+    }
+
+    case MSP430_SXT: {
+        return set_destination(op, NULL, SIGNED(16, get_destination(op)));
+    }
+    // TODO add the rest of instructions
+    default:
+        rz_warn_if_reached();
+        return rz_msp430_dummy();
+    }
+}
+
+RzILOpEffect *rz_msp430_lift_double_operand_instr(RZ_NONNULL RzAnalysis *analysis, const Msp430Instruction *op) {
+	return NOP(); // TODO
+}
+
+RzILOpEffect *rz_msp430_lift_jump_instr(RZ_NONNULL RzAnalysis *analysis, const Msp430Instruction *op) {
+	return NOP(); // TODO
+}
+
+RZ_IPI RzILOpEffect *rz_msp430_lift_instr(RZ_NONNULL RzAnalysis *analysis, const Msp430Instruction *op) {
```

Comment:
```suggestion
RZ_IPI RzILOpEffect *rz_msp430_lift_instr(RZ_NONNULL RzAnalysis *analysis, const Msp430Instruction *op) {
    rz_return_val_if_fail(analysis && op, NULL);
```

These kind of `NULL` checks at the beginning will help you a lot later to locate errors.
It is advisable to add them in every function.